descwl_coadd_task/tests/test_warps.py

```diff
@@ -348,6 +350,91 @@ def test_makeWarpMfrac(self, mask_bitname):
         wintrp = np.where(warp.mask.array & iflag != 0)
         self.assertEqual(wintrp[0].size, nexpected)
 
+    def test_makeWarpCompare(self):
+        """
+        Test the results match what we get calling warp_exposures directly
+
+        We need a version of this that persists to disk as well, to ensure the
+        round trip preserves the data
+        """
+        from descwl_coadd import warp_exposures
+
+        config = MakeShearWarpConfig()
+
+        # this masks a single pixel, which then propagates to mfrac
+        # in a set of neighboring pixels
+        for exposureRef in self.dataRefs:
+            self._fill_exposure(
+                exposureRef.get(),
+                mask_pixel=True,
+                mask_bitname="CR",
+            )
+
+        # this warps all exposures onto a common image, so they must not
+        # overlap.  Our _make_data creates the images from CCD bounding boxes
+
+        makeWarp = MakeShearWarpTask(config=config)
+        result = makeWarp.run(
+            self.dataRefs,
+            skyInfo=self.skyInfo,
+            visit_summary=None,
+        )
+
+        coadd_bbox, coadd_wcs = self.skyInfo.bbox, self.skyInfo.wcs
+        for exposureRef in self.dataRefs:
+            exp = exposureRef.get()
+
+            data_id = exposureRef.dataId
+            seed = MakeShearWarpTask.get_seed_from_data_id(data_id)
+
+            rng = np.random.RandomState(seed + config.seed_offset)
+            warped_exposures = warp_exposures(
+                exp,
+                coadd_wcs,
+                coadd_bbox,
+                remove_poisson=config.remove_poisson,
+                rng=rng,
+                bad_mask_planes=config.bad_mask_planes,
+                verify=False,
+            )
+
+            # these warps are same size as big image in result
+            warp, noise_warp, mfrac_warp, _ = warped_exposures
+
+            # get portion that was written into
```

Comment:
We also need to test that the NaNs in both datasets are in the same spot. In principle this could pass if one is the subset of the other.

Reply:
The NaN do not match, I think because only "good" pixels are copied in the task.

https://github.com/LSSTDESC/descwl-coadd-task/blob/d191d1acde16cc624fceb5b2f0ee21202694582c/descwl_coadd_task/_warp_task.py#L344